allensdk/brain_observatory/behavior/stimulus_processing.py

```diff
@@ -181,6 +186,35 @@ def get_visual_stimuli_df(data, time):
                     "end_frame": epoch_end,
                     "time": time[epoch_start],
                     "duration": time[epoch_end] - time[epoch_start],  # this will always work because an epoch will never occur near the end of time
+                    "omitted": False,
                 })
 
-    return pd.DataFrame(data=visual_stimuli_data)
\ No newline at end of file
+    visual_stimuli_df = pd.DataFrame(data=visual_stimuli_data)
+
+    # Add omitted flash info:
+    omitted_flash_list = []
+    omitted_flash_frame_log = data['items']['behavior']['omitted_flash_frame_log']
+    for stimuli_group_name, omitted_flash_frames in omitted_flash_frame_log.items():
+
+        stim_frames = visual_stimuli_df['frame'].values
+        omitted_flash_frames = np.array(omitted_flash_frames)
+
+        #  Test offsets of omitted flash frames to see if they are in the stim log
+        offsets = np.arange(-3, 4)
+        offset_arr = np.add(np.repeat(omitted_flash_frames[:, np.newaxis], offsets.shape[0], axis=1), offsets)
+        matched_any_offset = np.any(np.isin(offset_arr, stim_frames), axis=1)
+
+        #  Remove omitted flashes that also exist in the stimulus log
+        was_true_omitted = np.logical_not(matched_any_offset)  # bool
+        omitted_flash_frames_to_keep = omitted_flash_frames[was_true_omitted]
+
+        # Have to remove frames that are double-counted in omitted log
+        omitted_flash_list += list(np.unique(omitted_flash_frames_to_keep))
+
+    omitted = np.ones_like(omitted_flash_list).astype(bool)
+    time = [time[fi] for fi in omitted_flash_list]
+    omitted_df = pd.DataFrame({'omitted': omitted, 'frame': omitted_flash_list, 'time': time})
```

Comment:
If we don't give omitted stimuli string names for `image_name` and `image_set`, then those columns in the dataframe after we concatenate will contain a mixture of strings (from the visual stimuli) and `nan` values. This forces pandas to treat these as Object dtype columns. See: https://gist.github.com/nickponvert/3c1682b2431a0de35b1b65ffeb282bda

We should give omitted stimuli strings for their name and image set to avoid this.